server/public/model/temporary_post.go

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+import "errors"
+
+type TemporaryPost struct {
+	ID       string      `json:"id"`
+	Type     string      `json:"type"`
+	ExpireAt int64       `json:"expire_at"`
+	Message  string      `json:"message"`
+	FileIDs  StringArray `json:"file_ids"`
+}
+
+func (o *TemporaryPost) PreSave() error {
```

Comment:
This looks like `IsValid` rather than `PreSave`.  `PreSave` would set the ID if not already set like other `PreSave` methods.

Reply:
Makes sense 👍